### PR TITLE
[WIP] Resolved Cmake Config issues for binary generation

### PR DESCRIPTION
**Pull request by inbangsa**
This PR is created for testing the Github Actions, still in testing stage

### CODE REVIEW

File: cmake/modules/ifm3d-config.cmake.in [hunk 1]
```diff
@@ -27,6 +27,13 @@ else()
           else()
             find_package(OpenCV REQUIRED COMPONENTS core)
           endif()
+		  find_package(OpenCV 3 REQUIRED COMPONENTS core)
```
Comment (graugans) [inline cmake/modules/ifm3d-config.cmake.in:30]: Are you sure that we should look for OpenCV 3 here? I would have expected that this is captured by the if above. BTW, do you know where OpenCV 2.4 is used? If this is Ubuntu 14.04 or 16.04 we probably can remove the whole check.

File: modules/image/src/libifm3d_image/CMakeLists.txt [hunk 2]
```diff
@@ -56,11 +63,14 @@ target_link_libraries(
...
     ifm3d_framegrabber
     ifm3d_camera
-    ${OpenCV_LIBRARIES}
+    ${OpenCV_LIBS}
+    pcl_headers
   PRIVATE
     ${LIB_glog}
   )
 
+
```
Comment (graugans) [inline modules/image/src/libifm3d_image/CMakeLists.txt:72]: I guess the extra newlines can be removed

File: modules/swupdater/src/libifm3d_swupdater/CMakeLists.txt [hunk 3]
```diff
@@ -24,7 +24,6 @@ target_include_directories(ifm3d_swupdater
     $<INSTALL_INTERFACE:include>
     $<BUILD_INTERFACE:${IFM3D_SWUPDATER_BINARY_DIR}/include>
     $<BUILD_INTERFACE:${IFM3D_CAMERA_BINARY_DIR}/include>
-    ${CURL_INCLUDE_DIRS}
```
Comment (graugans) [inline modules/swupdater/src/libifm3d_swupdater/CMakeLists.txt:27]: good catch

**Review by graugans — CHANGES_REQUESTED** [review]

File: cmake/modules/ifm3d-config.cmake.in (hunk 1 above)
Comment (graugans) [reply cmake/modules/ifm3d-config.cmake.in:30]: Looks like Action responded already :-)

```
CMake Error at modules/image/src/libifm3d_image/CMakeLists.txt:9 (find_package):
  Could not find a configuration file for package "OpenCV" that is compatible
  with requested version "3.0".

  The following configuration files were considered but not accepted:

    /usr/share/OpenCV/OpenCVConfig.cmake, version: 2.4.9.1



-- Configuring incomplete, errors occurred!
See also "/home/runner/work/ifm3d/ifm3d/build/CMakeFiles/CMakeOutput.log".
See also "/home/runner/work/ifm3d/ifm3d/build/CMakeFiles/CMakeError.log".
Error: Process completed with exit code 1.
```

Comment (graugans) [reply cmake/modules/ifm3d-config.cmake.in:30]: For the backlog we should also add Ubuntu 20.04 to this list: https://github.com/ifm/ifm3d/blob/master/.github/workflows/ubuntu.yml#L15

**Comment by inbangsa** [comment]
Seems an issue with PCL and boost on Windows..

**Pull request closed without merging**